core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1094,11 +1111,11 @@ public IndexShardCacheEntity(IndexShard indexShard, QueryPhase queryPhase, Searc
              * results.*/
             final int expectedSizeInBytes = 512;
```

Comment:
`BytesStreamOutput` is pretty good at resizing I think we are fine here. The other question is if we should raise the overall size of the case since we now also cache field stats?